hathor/manager.py

```diff
@@ -133,7 +138,7 @@ def __init__(self, reactor: IReactorCore, peer_id: Optional[PeerId] = None, netw
 
         # XXX Should we use a singleton or a new PeerStorage? [msbrogli 2018-08-29]
         self.pubsub = pubsub or PubSubManager(self.reactor)
-        self.tx_storage = tx_storage or TransactionMemoryStorage()
+        self.tx_storage = tx_storage or TransactionMemoryStorage(with_all_index=enable_sync_v1)
```

Comment:
The memory-storage? I think it's still has some use, it is not the default anymore though. I think CompactStorage (json files) is less useful since it can't even hold the current mainnet unless using a filesystem tuned to handled the number of inodes (or a filesystem that doesn't use inodes, like btrfs). The memory storage is useful for tests and platforms where rocksdb isn't supported (currently Windows, but maybe it won't work on M1 Macs for example, this has never been tested).

So at the very least, I'd prefer to have the rocksdb and memory storage, I completely agree we could remove all the other storages.

Reply:
I mean to remove as default choice in `HathorManager` in case a tx_storage is not provided. Maybe we should make this a required parameter with no default choice.

Reply:
I think we can remove it as a default choice on `HathorManager`. But on `rocksdb` being required, I agree that for any real deploy that makes sense, but it might still be useful to be able to have the option to use a memory backend for every case where we use rocksdb, even if just for mocking, or testing on platforms that are tricky to install rocksdb into.

Reply:
I removed it as a implicit default on `HathorManager`. What do you think?

Reply:
Great. Thanks!